Jenkinsfile

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env groovy
+
+pipeline {
+    agent { docker defaultWorker.getConfig("openstack-test") }
+    options {
+        ansiColor('xterm')
+        timestamps()
+        timeout(time: 2, unit: "HOURS")
+    }
+    stages {
+        stage("systest") {
+            steps {
+                sh '''
+                    # - initialize job control vars
+                    # - JOB_NAME is an environment variable that is
+                    #   automatically provided by Jenkins
+                    #   (eg. "openstack/agent/mitaka/12_1_1-overcloud")
+                    export PROGRAM=$(echo $JOB_NAME | cut -d "/" -f 1)
+                    export PROJECT=$(echo $JOB_NAME | cut -d "/" -f 2)
+                    export BRANCH=$(echo $JOB_NAME | cut -d "/" -f 3)
+                    export SYSTEST_JOB=$(echo $JOB_NAME | cut -d "/" -f 4)
+
+                    # - print build properties
+                    printenv | sort | grep -v OS_PASSWORD
+
+                    # - setup ssh agent
+                    eval $(ssh-agent -s)
+                    ssh-add
+
+                    # - run tests
+                    if [ "$SYSTEST_JOB" = "unit-tests" ]; then
+                        make -C systest unit-tests
+                    else
+                        export TESTENV_PATH="/usr/local/share/testenv"
+                        export BIGIP=$(echo $SYSTEST_JOB | cut -d "-" -f 1)
+                        export CLOUD=$(echo $SYSTEST_JOB | cut -d "-" -f 2)
```

Comment:
I personally have no near-future plans to change the mechanism for passing input variables to the Jenkinsfile (mostly because the two or three methods I've already experimented with seemed like they were too heavy for the current needs).  But as I mentioned in the comment above, you guys are certainly free to move to properties files if/when you want/need them (and in that scenario, you might want to keep it simple and make your job name match the name of the associated properties file).